kernel/enter.c

```diff
@@ -820,6 +939,104 @@ int myst_enter_kernel(myst_kernel_args_t* args)
     if (__myst_kernel_args.perf)
         _print_boottime();
 
+    // open the appenv.json
+    char* data;
+    size_t size;
+    char* path = "appenv.json";
+    if (myst_load_file(path, (void**)&data, &size) != 0)
+    {
+        goto skip_docker_env;
+    }
+
+    // read all the env vars and values
+    json_parser_t parser;
+    json_result_t r;
+    json_callback_data_t callback_data;
+    static json_allocator_t allocator = {
+        malloc,
+        free,
+    };
+
+    memset(&callback_data, 0, sizeof(json_callback_data_t));
+    myst_args_init(&callback_data.env);
+
+    const json_parser_options_t options = {1};
+
+    if ((r = json_parser_init(
+             &parser,
+             data,
+             size,
+             _json_read_callback,
+             &callback_data,
+             &allocator,
+             &options)) != JSON_OK)
+    {
+        myst_eprintf("json_parser_init() failed: %d\n", r);
```

Comment:
Shouldn't we return on failure?

Reply:
Added an ERAISE to propogate the failure